cunumeric/linalg/linalg.py

```diff
@@ -75,7 +83,7 @@ def cholesky(a):
 
 # This implementation is adapted closely from NumPy
 @add_boilerplate("a")
-def matrix_power(a, n):
+def matrix_power(a: ndarray, n: int) -> Union[ndarray, None]:
```

Comment:
I think the return type should be just `ndarray`. Add a `None` check if you need it. Note that the while loop at line 152 always runs at least once.

Reply:
https://github.com/nv-legate/cunumeric/pull/387/commits/fdf2eee4b2a451ff3aa997d144fa6a3b9be40dc4